### ISSUE TITLE
[Bug]: Black screen at the beginning of the generation

### ISSUE THREAD

**Issue by Astarot11** [lshqqytiger/stable-diffusion-webui-directml#73]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

I installed SD using standart guide (https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Install-and-Run-on-AMD-GPUs) for Windows and AMD. Two days SD worked correctly. I created several images, upscaled them etc. It worked fine, my card (RX 580 8 gb) had only ~60% vram load and certanly didn't overheat. But today is different. Black screen appearing at the beginning of the generation. Computer doesn't turn off, I can still hear music. After the reboot, I saw that the video card in the device manager is disabled. It seems some error causes the video card to turn off. I hope there is some solution. 

### Steps to reproduce the problem

1. Write some prompt
2. Press generate
3. Enjoy the black screen

### What should have happened?

Just normal work of SD as it was earlier

### Commit where the problem happens

commit: ae337fa3

### What platforms do you use to access the UI ?

Windows

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
set COMMANDLINE_ARGS=--opt-sub-quad-attention --lowvram --disable-nan-check --autolaunch
```

### List of extensions

No

### Console logs

```Shell
venv "H:\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Commit hash: ae337fa39b6d4598b377ff312c53b14c15142331
Installing requirements for Web UI
Launching Web UI with arguments: --opt-sub-quad-attention --lowvram --disable-nan-check --autolaunch
Warning: experimental graphic memory optimization is disabled due to gpu vendor. Currently this optimization is only available for AMDGPUs.
Disabled experimental graphic memory optimizations.
Interrogations are fallen back to cpu. This doesn't affect on image generation. But if you want to use interrogate (CLIP or DeepBooru), check out this issue: https://github.com/lshqqytiger/stable-diffusion-webui-directml/issues/10
Warning: caught exception 'Torch not compiled with CUDA enabled', memory monitor disabled
No module 'xformers'. Proceeding without it.
Loading weights [6ce0161689] from H:\stable-diffusion-webui-directml\models\Stable-diffusion\v1-5-pruned-emaonly.safetensors
Creating model from config: H:\stable-diffusion-webui-directml\configs\v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Applying sub-quadratic cross attention optimization.
Textual inversion embeddings loaded(0):
Model loaded in 11.5s (load weights from disk: 0.4s, create model: 0.5s, apply weights to model: 10.0s, apply half(): 0.6s).
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 23.4s (import torch: 3.4s, import gradio: 2.6s, import ldm: 1.0s, other imports: 2.8s, setup codeformer: 0.2s, load scripts: 0.9s, load SD checkpoint: 11.7s, create ui: 0.3s, gradio launch: 0.3s).
```

### Additional information

I dont like these lines in console: "Warning: experimental graphic memory optimization is disabled due to gpu vendor. Currently this optimization is only available for AMDGPUs.". It's like the program can't detect my graphics card. Although it is definitely on, checked AMD Adrenaline UI, tried games etc.

**Comment by lshqqytiger** [comment]
`Warning: experimental graphic memory optimization is disabled due to gpu vendor. Currently this optimization is only available for AMDGPUs.`
This warning means that DirectML failed to detect your RX 580.
I'm not sure whether there is any damage on it or not.
Re-attach your RX 580 (I don't know what expression I should use. re-attach? re-establish? re-connect? I'm not English native) and check your softwares detect it well.

**Comment by Astarot11** [comment]
> > So should I try after starting Stable Diffusion switch model to some another? And then trying to generate picture.
> 
> yes, that work on my

Unfortunately, it doesnt work for me. I dont understand why SD was working earlier, but now it isn't working.

**Comment by Astarot11** [comment]
> `Warning: experimental graphic memory optimization is disabled due to gpu vendor. Currently this optimization is only available for AMDGPUs.` This warning means that DirectML failed to detect your RX 580. I'm not sure whether there is any damage on it or not. Re-attach your RX 580 (I don't know what expression I should use. re-attach? re-establish? re-connect? I'm not English native) and check your softwares detect it well.

I tried to re-attach, but it doesn't work either. I'm sure, that my RX580 works correctly, because it detects by AMD Software: Adrenaline edition. And i can play games normally, using resources of my card.

**Comment by lshqqytiger** [comment]
Now you can get error message and traceback.
Please pull latest commit and copy-paste them here.

**Comment by www54500** [comment]
I've also encountered the black screen error before, and I resolved it by underclocking the GPU

**Comment by lshqqytiger** [comment]
> I've also encountered the black screen error before, and I resolved it by underclocking the GPU

Yes. The main reason of black screen / accidentally shutdown / blue screen / etc... is overclocking or undervolting.
I have no idea factory overclocking can occur these issues, but you can try underclocking gpu.

Astarot11 // I pushed new commit. I could not test it because I'm outside now, but I hope it works for you.

**Comment by Astarot11** [comment]
Finally I made it work. But it took some time. First I reinstalled SD, error about memory disappeared. Second, I have done some underclocking. And it worked.... But black pictures were generated. I added some arguments such as --precision full --no-half --opt-split-attention-v1 and it worked! Finally. And even with underclocking, i removed --lowvram argument. Now my RX580 uses all memory with ease and the generation is done very quickly. 

The main thing - it works now. Thank you all for fixes and advices! Though I still don't understand why it worked earlier, when I just installed it.
